storage/tianmu/core/query_compile.cpp

```diff
@@ -300,17 +300,26 @@ bool Query::FieldUnmysterify(Item *item, TabID &tab, AttrID &col) {
 
 QueryRouteTo Query::AddJoins(List<TABLE_LIST> &join, TabID &tmp_table, std::vector<TabID> &left_tables,
                              std::vector<TabID> &right_tables, bool in_subquery, bool &first_table /*= true*/,
-                             bool for_subq_in_where /*false*/) {
+                             bool for_subq_in_where /*false*/, bool use_tmp_when_no_join /*false*/) {
   // on first call first_table = true. It indicates if it is the first table to
   // be added is_left is true iff it is nested left join which needs to be
   // flatten (all tables regardless of their join type need to be left-joined)
   TABLE_LIST *join_ptr;
   List_iterator<TABLE_LIST> li(join);
   std::vector<TABLE_LIST *> reversed;
 
-  if (!join.elements)
-    return QueryRouteTo::kToMySQL;  // no tables in table list in this
-                                    // select
+  if (!join.elements) {
+    if (use_tmp_when_no_join) {
+      TabID tab(-1);

```

Comment:
we dont know the meaning of `use_tmp_when_no_join `, pls give some explainations on this variable.